File: django_app/iss/iss/surveys/surveyWidget.py
from iss.surveys.abstractWidgetContainerSubclass import AbstractWidgetContainerSubclass


class SurveyWidget(AbstractWidgetContainerSubclass):
    args_data = {
        'name': {
            'type': 'string',
            'required': 0,
            'default': 'undefined'
        },
        'destination': {
            'type': 'string',
            'required': 0,
            'default': 'undefined'
        },
        'container': {
            'type': 'string',
            'required': 0,
            'default': '"#image .panel-body"'
        }
    }

    unnamed_args = ['name', 'destination', 'container']

File: django_app/iss/iss/static/surveys/js/surveyWidget.js
iss.lib.widgets.SurveyWidget = function(options) {
    return function(condition) {
        if (condition()) {
            return function() {
                return {
                        "destination": options.destination(),
                        "name": options.name(),
                        "container": options.container()
                    };
            }
        }
        else {
            return function() {
                return null;
            }
        }
    }
}
